sdk/ml/azure-ai-ml/azure/ai/ml/entities/_schedule/schedule.py

```diff
@@ -250,11 +254,18 @@ def _to_rest_object(self) -> RestSchedule:
 
         :return: Rest schedule.
         """
+        if isinstance(self.create_job, BaseNode):
+            self.create_job = self.create_job._to_job()
         if isinstance(self.create_job, PipelineJob):
             job_definition = self.create_job._to_rest_object().properties
             # Set the source job id, as it is used only for schedule scenario.
             job_definition.source_job_id = self.create_job.id
+        elif isinstance(self.create_job, CommandJob):
+            job_definition = self.create_job._to_rest_object().properties
+            # TODO: Merge this branch with PipelineJob after source job id move to JobBaseProperties
```

Comment:
What's this TODO for, the logic in both branches seems identical

Reply:
We have source_job_id only in PipelineJob now, and need to add it into JobBaseProperties so that all type of job could set that. the assignment statement in command job branch actually take no effect because rest model didn't source_job_id field